noir-projects/aztec-nr/aztec/src/state_vars/shared_immutable.nr

```diff
@@ -24,12 +24,24 @@ impl<T> StablePublicState<T> {
 
     // Intended to be only called once. 
     pub fn initialize<T_SERIALIZED_LEN>(self, value: T) where T: Serialize<T_SERIALIZED_LEN> {
-        assert(self.context.private.is_none(), "Public state wrties only supported in public functions");
-        // TODO: Must throw if the storage slot is not empty -> cannot allow overwriting
-        // This is currently impractical, as public functions are never marked `is_contract_deployment`
-        // in the `call_context`, only private functions will have this flag set.
-        let fields = T::serialize(value);
-        storage_write(self.storage_slot, fields);
+        assert(
+            self.context.private.is_none(), "SharedImmutable can only be initialized from public functions"
+        );
+        self.context.public.unwrap_unchecked().assert_deployment();
+
+        // We check that the struct is not yet initialized by checking if the placeholder slot is 0
+        let placeholder_storage_slot = pedersen_hash(
```

Comment:
Why do you want to compute it like this? 👀 It is public information anyway you could just use the value. I guess it is helping a little to make it behave somewhat similar to public state in the layout and macro, but otherwise not seeing a benefit

Reply:
Initially did that to get it to compile with the current macros but agree that it should be addressed in a less stupid way. Did that in https://github.com/AztecProtocol/aztec-packages/pull/4739/commits/165027da93b276d1848f86ffdc5c40901e54785b

Reply:
Re-worked in https://github.com/AztecProtocol/aztec-packages/pull/4739/commits/5ed77a47dc0a83f901143b0d0b87eefc444a6080